### PR TITLE
Loop instead of recursion

### PR DESCRIPTION
**Pull request by bugadani**
I saw the comment `// FIXME: consider not using recursion to lower this.` and considered not using recursion :)

### CODE REVIEW

File: compiler/rustc_ast_lowering/src/pat.rs [hunk 1]
```diff
@@ -12,80 +12,89 @@ use rustc_span::{source_map::Spanned, Span};
...
-                }
-                PatKind::Struct(ref path, ref fields, etc) => {
-                    let qpath = self.lower_qpath(
-                        p.id,
-                        &None,
-                        path,
-                        ParamMode::Optional,
-                        ImplTraitContext::disallowed(),
-                    );
+            let mut pattern = p;
```
Comment (nagisa) [inline compiler/rustc_ast_lowering/src/pat.rs:15]: minor style nit: you can make the argument itself `mut` instead of introducing a new binding.

**Comment by nagisa** [comment]
r? @nagisa r=me

Overall LGTM. Left a minor style nit but up to you if you want to address it.

@bors delegate=bugadani

File: compiler/rustc_ast_lowering/src/pat.rs (hunk 1 above)
Comment (bugadani) [reply compiler/rustc_ast_lowering/src/pat.rs:15]: I know, I intentionally didn't do that, though. And shot myself in the foot with that... So I think I'll take your advice, tomorrow :)

**Pull request merged**